zipline/finance/performance/tracker.py

```diff
@@ -189,170 +191,117 @@ def set_date(self, date):
+                 emission_type=None):
+        """
         Creates a dictionary representing the state of this tracker.
         Returns a dict object of the form described in header comments.
-        """
 
+        Use this method internally, when stats are available.
+        """
         # Default to the emission rate of this tracker if no type is provided
         if emission_type is None:
             emission_type = self.emission_rate
 
+        position_tracker = self.position_tracker
+
         _dict = {
             'period_start': self.period_start,
             'period_end': self.period_end,
             'capital_base': self.capital_base,
-            'cumulative_perf': self.cumulative_performance.to_dict(),
+            'cumulative_perf': self.cumulative_performance.to_dict(
+                pos_stats, cumulative_stats, position_tracker,
+            ),
             'progress': self.progress,
             'cumulative_risk_metrics': self.cumulative_risk_metrics.to_dict()
         }
         if emission_type == 'daily':
-            _dict['daily_perf'] = self.todays_performance.to_dict()
+            _dict['daily_perf'] = self.todays_performance.to_dict(
+                pos_stats,
+                todays_stats,
+                position_tracker)
         elif emission_type == 'minute':
             _dict['minute_perf'] = self.todays_performance.to_dict(
+                pos_stats,
+                todays_stats,
+                position_tracker,
                 self.saved_dt)
         else:
             raise ValueError("Invalid emission type: %s" % emission_type)
 
         return _dict
 
-    def _handle_event_price(self, event):
-        # updates last sale, and pays out a cash adjustment if applicable
-        cash_adjustment = self.position_tracker.update_last_sale(event)
-        if cash_adjustment != 0:
-            for perf_period in self.perf_periods:
-                perf_period.handle_cash_payment(cash_adjustment)
+    def copy_state_from(self, other_perf_tracker):
+        self.all_benchmark_returns = other_perf_tracker.all_benchmark_returns
 
-    def process_trade(self, event):
-        self._handle_event_price(event)
+        if other_perf_tracker.position_tracker:
+            self.position_tracker._unpaid_dividends = \
+                other_perf_tracker.position_tracker._unpaid_dividends
 
-    def process_transaction(self, event):
-        self._handle_event_price(event)
-        self.txn_count += 1
-        self.position_tracker.execute_transaction(event)
-        for perf_period in self.perf_periods:
-            perf_period.handle_execution(event)
-
-    def process_dividend(self, dividend):
+            self.position_tracker._unpaid_stock_dividends = \
+                other_perf_tracker.position_tracker._unpaid_stock_dividends
 
-        log.info("Ignoring DIVIDEND event.")
+    def process_transaction(self, transaction):
+        self.txn_count += 1
```

Comment:
I think it is only used for debug purposes, we may be able to remove it completely.